components/notebook-controller/config/overlays/openshift/generate-metadata-yaml.sh

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env bash
+
+set -uo pipefail
+
+function trap_exit() {
+	rc=$?
+
+	set +x
+
+	exit $rc
+}
+
+trap "trap_exit" EXIT
+
+_derive_metadata()
+{
+  # inspired from https://stackoverflow.com/a/29835459
+  current_dir=$(CDPATH= cd -- "$(dirname -- "$0")" && pwd)
+
+  kf_project_file="${current_dir}/PROJECT"
+  if [ -e "${kf_project_file}" ]; then
+
+    if [ -z "${metadata_repo_url}" ]; then
+      project_repo_reference=$(yq -e '.repo' "${kf_project_file}")
+      project_repo_parts=( $(printf "%s" ${project_repo_reference##https://} | tr '/' ' ') )
+      github_host="${project_repo_parts[0]}"
+      github_owner="${project_repo_parts[1]}"
+      github_repo="${project_repo_parts[2]}"
+
+      metadata_repo_url=$(printf "https://%s/%s/%s" "${github_host}" "${github_owner}" "${github_repo}")
+    fi
+
+    if [ -z "${metadata_name}" ]; then
```

Comment:
**`"Controversial" Decision #2`**

While the specification for `odh_metadata.yaml` states the `name` attribute should be a human-readable name.. it doesn't really provide any template/structure/consistency w.r.t how the `name` should be formulated.

With my intent to allow full automation w/out user input... I then opted to have the script generate the `name` based on information present in the `PROJECT` file ([as outlined above](https://github.com/red-hat-data-services/kubeflow/pull/172/files#r1856806738))

This would result in the following:
- `name: kubeflow.org notebook-controller`

I can change this default naming scheme based on preferences of team... but to automate generating the name (as presented in the Jira issue) of `Kubeflow Notebook` - that was going to require a lot more shell scripting that I felt was necessary/reasonable when the "core ask" here is from this ill-defined concept of "human readable".. Again happy to pivot!

Good idea? Bad Idea? _You tell me!_